libvast/vast/system/type_registry.hpp

```diff
@@ -64,9 +66,12 @@ struct type_registry_state {
   caf::error load_from_disk();
 
   /// Store a new layout in the registry.
-  void insert(vast::record_type layout);
+  void insert(vast::type layout);
 
   /// Get a list of known types from the registry.
+  std::unordered_set<vast::type> lookup() const;
```

Comment:
I know why you called this function `lookup`, but without a parameter, it's misleading. How about calling this function `types()` instead?